GoogleDataTransport/GDTCORLibrary/GDTCOREventMetricsCounter.m

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "GoogleDataTransport/GDTCORLibrary/Private/GDTCOREventMetricsCounter.h"
+
+#import "GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCOREvent.h"
+
+typedef NSDictionary<NSNumber *, NSNumber *> GDTCORDroppedEventCounter;
+
+@interface GDTCOREventMetricsCounter ()
+
+// TODO(ncooke3): Document.
+@property(nonatomic)
+    NSDictionary<NSString *, GDTCORDroppedEventCounter *> *droppedEventCounterByMappingID;
+
+@end
+
+@implementation GDTCOREventMetricsCounter
+
++ (instancetype)counterWithEvents:(NSArray<GDTCOREvent *> *)events
+                 droppedForReason:(GDTCOREventDropReason)reason {
+  NSMutableDictionary<NSString *, GDTCORDroppedEventCounter *> *eventCounterByMappingID =
+      [NSMutableDictionary dictionary];
+
+  for (GDTCOREvent *event in events) {
+    // TODO(ncooke3): Should I use an autorelease pool?
+    // Dropped events with a `nil` or empty mapping ID (log source) are not recorded.
+    if ([event.mappingID length] == 0) {
```

Comment:
nit: use dot property

Reply:
Resolved in [4a4fa4d](https://github.com/google/GoogleDataTransport/pull/53/commits/4a4fa4d90d4747d0aa338a5dbb353b8e2466fdd3)